test/integration-tests/testexplorer/utilities.ts

```diff
@@ -136,3 +133,170 @@ export function eventPromise<T>(event: vscode.Event<T>): Promise<T> {
         event(t => resolve(t));
     });
 }
+
+function decomposeSettingName(setting: string): { section: string; name: string } {
+    const splitNames = setting.split(".");
+    const name = splitNames.pop();
+    const section = splitNames.join(".");
+    if (name === undefined) {
+        throw new Error(`Invalid setting name: ${setting}, must be in the form swift.settingName`);
+    }
+    return { section, name };
+}
+
+export type SettingsMap = { [key: string]: unknown };
+
+/**
+ * Updates VS Code workspace settings and provides a callback to revert them. This
+ * should be called before the extension is activated.
+ *
+ * This function modifies VS Code workspace settings based on the provided
+ * `settings` object. Each key in the `settings` object corresponds to a setting
+ * name in the format "section.name", and the value is the new setting value to be applied.
+ * The original settings are stored, and a callback is returned, which when invoked,
+ * reverts the settings back to their original values.
+ *
+ * @param settings - A map where each key is a string representing the setting name in
+ * "section.name" format, and the value is the new setting value.
+ * @returns A function that, when called, resets the settings back to their original values.
+ */
+export async function updateSettings(settings: SettingsMap): Promise<() => Promise<void>> {
+    const applySettings = async (settings: SettingsMap) => {
+        const savedOriginalSettings: SettingsMap = {};
+        Object.keys(settings).forEach(async setting => {
+            const { section, name } = decomposeSettingName(setting);
+            const config = vscode.workspace.getConfiguration(section, { languageId: "swift" });
+            savedOriginalSettings[setting] = config.get(name);
+            await config.update(
+                name,
+                settings[setting] === "" ? undefined : settings[setting],
+                vscode.ConfigurationTarget.Workspace
+            );
+        });
+
+        // There is actually a delay between when the config.update promise resolves and when
+        // the setting is actually written. If we exit this function right away the test might
+        // start before the settings are actually written.
+        await new Promise(resolve => setTimeout(resolve, 500));
```

Comment:
Hmm, I wonder if doing a getConfiguration again will work better, and possibly a bit more deterministic.

Reply:
Do you mean in a loop to verify its set? If so that's reasonable.

Reply:
I honestly would have assume the await config.update then getConfiguration would work without polling. But if not yeah a more deterministic poll might be better.

Scarred by hand set delay in tests from experience. 😂

Reply:
I've refined it a bit to check in a loop, lets see how it goes.